torchrl/envs/transforms/transforms.py

```diff
@@ -3607,3 +3607,56 @@ def transform_input_spec(self, input_spec: CompositeSpec) -> CompositeSpec:
             if not self.create_copy:
                 del input_spec[in_key]
         return input_spec
+
+
+class Reward2GoTransform(Transform):
+    """Calculates the reward to go based on the episode reward and a discount factor.
+
+    As the Reward2GoTransform is only an inverse transform the in_keys will be directly used for the in_keys_inv.
+    The reward-to-go can be only calculated once the episode is finished. Therefore, the transform should be applied to the replay buffer
+    and not to the collector.
```

Comment:
makes me think:
when we register a transform to an env, we call `Transform.set_container`.
What we could do is this
```
def set_container(self, container):
    if isinstance(container, Envbase) or  container.parent is not None:
        raise ValueError(...)
```
That won't prevent someone from doing
```
t = Reward2GoTransform()
t = Compose(t)
env = TransformedEnv(env, t)
```
but it will prevent
```
env = TransformedEnv(env)
env.append_transform(t)
```

Reply:
added it and also added:

```
    def _apply_transform(self, reward: torch.Tensor) -> torch.Tensor:
        raise ValueError(
            "The Reward2GoTransform is only an inverse transform and can only be applied to the replay buffer and not to the collector or the environment."
        )
```
Now catching both cases or generally, any apply call.

Reply:
Okay, my fault when sampling from the replay buffer the apply_transform is called and would trigger the error. So it cant be there.